### PR TITLE
fix(biome_console): fix printing emojis consisting of multiple codepoints

### LINKED ISSUE #455: 🐛 `biome format` breaks emojis when used via stdin
### Environment information

```block
CLI:
  Version:                      1.2.2
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.7.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0

Discovering running Biome servers...

Server:
  Status:                       stopped
```

### What happened?

When using `biome format` via stdin, some emojis seem to break. This does not affect all emojis, and it does not affect `biome format --write`.

(The different emoji sizes is a Wezterm-font-thing, I checked that the issue persists when opening the file in TextEdit.)

Pasted image 2023-09-30 at 13 14 41@2x

### Expected result

Emojis not breaking

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

### PR DESCRIPTION
**Pull request by simonxabris**
## Summary

Fixes #455 

The source of the bug was, that the code used the `.chars()` iterator, which creates Unicode code point chunks. Some complex unicode characters, like emojis can consist of multiple code points, which the code did not handle correctly. As suggested in the PR (and the conclusion I came to as well), I used the `unicode_segmentation` library, to be able to iterate over graphemes (which is basically a character as perceived by humans). This fixes the error by treating multiple code points representing a single character as one unit.

## Test Plan

I've added a new test where the input string contains emojis that the CLI couldn't print before, and I'm basically asserting that the output is the exact same as the input.

### CODE REVIEW

**Comment by simonxabris** [comment]
I see from the test run the that windows character replacement stuff is not working fully, I'll try to figure out something there!

File: crates/biome_console/src/write/termcolor.rs [hunk 1]
```diff
@@ -145,31 +146,67 @@ where
...
         }
 
         Ok(())
     }
 }
 
+/// Determines if a unicode grapheme consists only of code points
+/// which are considered whitepsace characters in ASCII
+fn grapheme_is_whitespace(grapheme: &str) -> bool {
+    grapheme.chars().all(|c| c.is_whitespace())
```
Comment (ematipico) [inline crates/biome_console/src/write/termcolor.rs:201]: TIL "grapheme" 😄

File: CHANGELOG.md [hunk 2]
```diff
@@ -42,6 +42,8 @@ Read our [guidelines for writing a good changelog entry](https://github.com/biom
 
 - Fix [#676](https://github.com/biomejs/biome/issues/676), by using the correct node for the `"noreferrer"` when applying the code action. Contributed by @ematipico
 
+-Fix [#455](https://github.com/biomejs/biome/issues/455). The CLI can now print complex emojis to the console correctly.
```
Comment (ematipico) [inline CHANGELOG.md:45]:
```suggestion
- Fix [#455](https://github.com/biomejs/biome/issues/455). The CLI can now print complex emojis to the console correctly.
```

File: website/src/content/docs/internals/changelog.mdx [hunk 3]
```diff
@@ -48,6 +48,8 @@ Read our [guidelines for writing a good changelog entry](https://github.com/biom
 
 - Fix [#676](https://github.com/biomejs/biome/issues/676), by using the correct node for the `"noreferrer"` when applying the code action. Contributed by @ematipico
 
+-Fix [#455](https://github.com/biomejs/biome/issues/455). The CLI can now print complex emojis to the console correctly.
```
Comment (ematipico) [inline website/src/content/docs/internals/changelog.mdx:51]:
```suggestion
- Fix [#455](https://github.com/biomejs/biome/issues/455). The CLI can now print complex emojis to the console correctly.
```

**Review by ematipico — APPROVED** [review]
And here I thought that the issue was the formatter :) Great work here and thank you for fixing it!

**Pull request merged**